src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/SubscriptionFactory.cs

```diff
@@ -38,7 +40,8 @@ public SubscriptionFactory(ILogManager? logManager,
         IFilterStore? filterStore,
         IEthSyncingInfo ethSyncingInfo,
         ISpecProvider specProvider,
-        IJsonSerializer jsonSerializer)
+        IJsonSerializer jsonSerializer,
+        IPeerPool peerPool)
```

Comment:
Maybe instead of passing `IPeerPool` here and other specialized types from different modules, we should register subscriptions from the outside of this class?